### ISSUE TITLE
Ping Timeout Issue - Destination Host Unreachable

### ISSUE THREAD

**Issue by kiwisincebirth** [cdarras/clamav-client#14]
This is a comment for improvement based on the following fairly common scenario. 

User uploads a document to our web application, and the upload is BLOCKED if the file is flagged by Clam AV . HOWEVER if the ClamAV service is OFFLINE, we DONT BLOCK the upload operation, (bad user experience) we allow the upload to continue, and flag the file as requiring a seperate scan at a latter date

To achieve this we need a way of determining if ClamAV is available. The in built ping() command, unfortunately has a long timeout (> 1 minute) when the destination host is unreadable. To overcome this. the following code was used. It would be better if tis code was made part of (at least the function) the the ClamAV client library

```
fun ClamavClient.isReachable(timeout: Int) : Boolean {
    return try {
        // perform an independent socket connection with a connection timeout.
        return SelectorProvider.provider().openSocketChannel().use {
            it.configureBlocking(true)
            it.socket().use {
                it.connect(this.server, timeout)
                true
            }
        }
    } catch (e: SocketTimeoutException) {
        false
    }
}
```

**Comment by cdarras** [comment]
Thanks! Available in 2.1.1-SNAPSHOT

**Comment by kiwisincebirth** [comment]
Glad to see you implemented this.
